src/main/java/org/hiero/sketch/spreadsheet/QuantileSketch.java

```diff
@@ -0,0 +1,253 @@
+     *
+     * @param resolution Number of buckets: percentiles correspond to 100 buckets etc.
+     * @return A table of size resolution, whose ith entry is ranked approximately i/resolution.
+     */
+    public QuantileList getQuantile(final Table data, final int resolution) {
+        /* Sample a set of rows from the table. */
+        final int perBin = 100;
+        final int numSamples = resolution * perBin;
+        final int dataSize = data.getNumOfRows();
+        final IMembershipSet sampleSet = data.members.sample(numSamples);
+        final Table sampleTable = data.compress(sampleSet);
+        /* Sort the sampled rows. */
+        final int realSize = sampleTable.getNumOfRows();
+        final Integer[] order = new Integer[realSize];
+        for (int i = 0; i < realSize; i++) {
+            order[i] = i;
+        }
+        Arrays.sort(order, this.getComparator(sampleTable));
+        /* Pick equally spaced elements as the sample quantiles.*/
+        final int[] quantile = new int[resolution + 1];
+        final ApproxRank[] approxRank = new ApproxRank[resolution + 1];
+        for (int i = 0; i <= resolution; i++) {
+            quantile[i] = order[((realSize - 1) * i) / resolution];
+            final int step = dataSize / resolution;
+            approxRank[i] = new ApproxRank(i * step, (resolution - i) * step);
+        }
+        final IRowOrder quantileMembers = new ArrayRowOrder(quantile);
+        final HashSubSchema subSchema = new HashSubSchema();
+        for (final OrderedColumn ordCol : this.sortOrder) {
+            subSchema.add(ordCol.colName);
+        }
+        return new QuantileList(sampleTable.compress(subSchema, quantileMembers),
+                approxRank, dataSize);
+    }
+
+    private boolean[] getMergeOrder(@NonNull final QuantileList left, @NonNull final QuantileList right) {
+        if (!left.getSchema().equals(right.getSchema())) {
+            throw new RuntimeException("Tables do not have matching schemas");
+        }
+        final int length = left.getQuantileSize() + right.getQuantileSize();
+        final boolean[] mergeLeft = new boolean[length];
+        int i = 0, j = 0, k = 0;
+        int outcome;
+        while ((i < left.getQuantileSize()) && (j < right.getQuantileSize())) {
+            outcome = 0;
+            for (final OrderedColumn ordCol : this.sortOrder) {
+                final IColumn leftCol = left.getColumn(ordCol.colName);
+                final IColumn rightCol = right.getColumn(ordCol.colName);
+                final boolean leftMissing = leftCol.isMissing(i);
+                final boolean rightMissing = rightCol.isMissing(j);
+                if (leftMissing && rightMissing) {
+                    outcome = 0;
+                } else if (leftMissing) {
+                    outcome = 1;
+                } else if (rightMissing) {
+                    outcome = -1;
+                } else {
+                    switch (left.getSchema().getKind(ordCol.colName)) {
```

Comment:
Please add a // TODO here: we should move this into a separate class that has static methods and is used to compare "Objects" that reside in columns.